src/components/axes/timeAxis.ts

```diff
@@ -207,6 +202,19 @@ export module Axis {
         return this._possibleTimeAxisConfigurations;
       }
       this._possibleTimeAxisConfigurations = configurations;
+      this._numTiers = _Util.Methods.max(this._possibleTimeAxisConfigurations.map((config: TimeAxisConfiguration) => config.length), 0);
+
+      if (this._isAnchored) {
+        this._createDomElements();
+      }
+
+      var oldLabelPositions: string[] = this.tierLabelPositions();
+      var newLabelPositions: string[] = [];
+      for (var i = 0; i < this._numTiers; i++) {
+        newLabelPositions.push(oldLabelPositions[i] || "between");
+      }
+      this.tierLabelPositions(newLabelPositions);
```

Comment:
Ok I see what you are getting at.

Can we remove the newLabelPositions variable?  I don't think that variable is needed.  Also, if you remove that variable, can you remove the `old` from `oldLabelPositions`